app/services/spree_adyen/gateways/configure.rb

```
@@ -0,0 +1,59 @@
+        webhook_url = URI.parse(Spree::Core::Engine.routes.url_helpers.adyen_webhooks_url(host: gateway.stores.first.url))
+        webhook_url.scheme = 'https'
+        set_up_webhook_request = gateway.set_up_webhook(webhook_url.to_s)
+        return unless set_up_webhook_request.success?
+
+        gateway.preferred_webhook_id = set_up_webhook_request.authorization
+        generate_hmac_key_request = gateway.generate_hmac_key
+        return unless generate_hmac_key_request.success?
+
+        gateway.previous_hmac_key = gateway.preferred_hmac_key if gateway.preferred_hmac_key.present?
+        gateway.preferred_hmac_key = generate_hmac_key_request.authorization
+      end
```

RuboCop (1.76.1)

[convention] Assignment Branch Condition size for `set_up_webhook_with_hmac_key` is too high. [<7, 29, 3> 29.98/17]

(Metrics/AbcSize)